src/main/java/hudson/plugins/active_directory/ActiveDirectoryAuthenticationProvider.java

```diff
@@ -99,63 +121,80 @@ static String dnToLdapUrl(String dn) {
+                        try {
+                            usr = (authentication==null
+                                    ? dso.openDSObject(dnToLdapUrl(dn), null, null, ADS_READONLY_SERVER)
+                                    : dso.openDSObject(dnToLdapUrl(dn), dn, password, ADS_READONLY_SERVER))
+                                    .queryInterface(IADsUser.class);
+                        } catch (ComException e) {
+                            // this is failing
+                            String msg = String.format("Incorrect password for %s DN=%s: error=%08X", username, dn, e.getHRESULT());
+                            LOGGER.log(Level.FINE, String.format("Login failure: Incorrect password for %s DN=%s: error=%08X", username, dn, e.getHRESULT()), e);
+                            throw (BadCredentialsException)new BadCredentialsException(msg).initCause(e);
+                        }
+                        if (usr == null)    // the user name was in fact a group
+                            throw new UsernameNotFoundException("User not found: "+ username);
+
+                        List<GrantedAuthority> groups = new ArrayList<GrantedAuthority>();
+                        for( Com4jObject g : usr.groups() ) {
+                            if (g==null) {
+                                continue;   // according to JENKINS-17357 in some environment the collection contains null
+                            }
+                            IADsGroup grp = g.queryInterface(IADsGroup.class);
+                            // cut "CN=" and make that the role name
+                            groups.add(new GrantedAuthorityImpl(grp.name().substring(3)));
+                        }
+                        groups.add(SecurityRealm.AUTHENTICATED_AUTHORITY);
+
+                        LOGGER.log(Level.FINE, "Login successful: {0} dn={1}", new Object[] {username, dn});
+
+                        return new ActiveDirectoryUserDetail(
+                                username, password,
+                                !isAccountDisabled(usr),
+                                true, true, true,
+                                groups.toArray(new GrantedAuthority[groups.size()]),
+                                getFullName(usr), getEmailAddress(usr), getTelephoneNumber(usr)
+                        ).updateUserInfo();
+                    } finally {
+                        col.disposeAll();
+                        COM4J.removeListener(col);
+                    }
+                }
+            });
+        } catch (UncheckedExecutionException e) {
+            Throwable t = e.getCause();
+            if (t instanceof AuthenticationException) {
+                AuthenticationException authenticationException = (AuthenticationException)t;
+                throw authenticationException;
+            } else {
+                throw new GuavaAuthenticationException("Authentication failed because there was a problem caching user " + username);
```

Comment:
I'd go for the existing `AuthenticationException` one with the caught one as the cause.